platform/platform-impl/src/com/intellij/openapi/fileEditor/impl/FileOffsetsManager.java

```diff
@@ -59,6 +59,11 @@ public int getOriginalOffset(@NotNull final VirtualFile file, final int converte
     return getCorrespondingOffset(offsets.myConvertedLineOffsets, offsets.myOriginalLineOffsets, convertedOffset);
   }
 
+  public int getLineNumber(@NotNull final VirtualFile file, final int offset) {
+    final LineOffsets offsets = getLineOffsets(file);
+    return -Arrays.binarySearch(offsets.myOriginalLineOffsets, offset) - 1;
```

Comment:
I guess the right code would be 
```
    int line = Arrays.binarySearch(offsets.myOriginalLineOffsets, originalOffset);
    return line >= 0 ? line : -line - 2;
```
so that it behaves similar to `Document.getLineNumber()`.